docs/programming.md

```diff
@@ -0,0 +1,75 @@
+
+All languages listed here come with examples to help you get started.
+
+There are several getting started guides on the internet too, as well as YouTube videos, such as those from [https://www.youtube.com/@AgonBits] 
+
+
+
+## The Languages
+
+### Basic
+
+This most popular language is installed by many Agon owners and is a great way to quickly start programming your Agon computer. 
+
+The Agon port of BBC Basic provides a familiar yet powerful language used by beginners and experts alike. It runs directly on the Agon computer and is often started automaticaly by many users with a command in the `autoexec.txt` file.
+
+Download from here (16bit mode): https://github.com/breakintoprogram/agon-bbc-basic
+
+Download from here (ADL mode): https://github.com/breakintoprogram/agon-bbc-basic-adl
+
+
+### Z80 Assembly
+
+If you have ever written software in z80 Assembly for the Sinclair ZX81, ZX Spectrum, MSX, Amstrad CPC, TRS-80, Nintendo Game Boy, or even the Texas Instruments calulators, then this will be familliar territory for you.
+
+You can write your code in whatever text editor you like, then assmble your source code with `ez80asm`, a z80 assembler which runs on the Agon itself, or on Mac, Windows, Linux, and even Raspberry Pi.
+
+Download ez80asm assembler from here: https://github.com/AgonPlatform/agon-ez80asm
+
+
+### C++
+
+Many programmers enjoy coding in `C` or `C++`. This language is also available and there are a few options for coding on your desktop PC, then compiing and transferring onto your Agon to run.
+
+A whole development toolchain is available for Windows called `AgDev`, and a Mac/Linux/Raspberry Pi toolchain called `Agondev`.
+
+Download Windows (AgDev) from here: https://github.com/pcawte/AgDev
+
+And Mac/Linux/Raspberry Pi (AgonDev) from here: https://github.com/AgonPlatform/agondev
+
+
+### B Simple
+
+_B_ was a language which, unsurpisingly, was around before _C_. This interpretation, `B Simple`, allows you to create programs that will be familliar to anyone who has used C, but without all the advanced features of C++. B Simple can be used on the Agon itslef, or on your desktop PC under Mac, Linux, Windows or Raspberry Pi. 
+
+`B Simple`'s approach is to compile your code into z80 assembler, then use the power of `ez80asm` to assemble the compiled code into the final binary executable. It can eaasily be expanded by writing custom libraries in z80 assembler.
+
+Download B Simple from here: https://github.com/nihirash/bsimple-ez80-compiler
+
+### Forth
+
+Some Agon users might reminisce on the days long gone of owning (or wishing they owned) a Jupiter Ace, with its more obscure _Forth_ programming language. 
```

Comment:
hehe :grin:

Forth isn't actually that obscure.  it has quite a wide usage history in embedded systems.  the Open Firmware boot ROM system (used by Apple, IBM, SUN and others) used Forth for it's CLI.

"niche" would maybe be a better descriptor than "obscure" 😀